### PR TITLE
Model of single multiple questions 

### PR DESCRIPTION
**Pull request by viharsanghavi**
Redmine Issue no: #20894
File changes are:
TrappistDBContext.cs
QuestionBase.cs
SingleMultipleAnswerQuestion.cs
SingleMultipleAnswerQuestionOption.cs
EnumList.cs

### CODE REVIEW

File: Trappist/global.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿{
   "projects": [ "src", "test" ],
   "sdk": {
-    "version": "1.0.0-preview2-1-003177"
+    "version": "1.0.0-preview2-003131"
```
Comment (Krunal-Promact) [inline Trappist/global.json]: why did you change the version to an older one?

File: Trappist/src/Promact.Trappist.DomainModel/Models/Enums/EnumList.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Promact.Trappist.DomainModel.Models.Enums
+{
+    public class EnumList
+    {
+        public enum QuestionType
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Enums/EnumList.cs]: please add individual enums. do not create enum list.

Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Enums/EnumList.cs]: What is the meaning of Programming enum?

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionBase.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    public class QuestionBase:BaseModel
+    {
+        [Required]
+        public string QuestionDetail { get; set; }
+        [Required]
+        public QuestionType QuestionType { get; set; }
+        [Required]
+        public DifficultyLevel DifficultyLevel { get; set; }
+        [Required]
+        public int CreateBy { get; set; }
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionBase.cs]: Why CreateBy is int?

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/SingleMultipleAnswerQuestion.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace Promact.Trappist.DomainModel.Models.Question
+{
+    public class SingleMultipleAnswerQuestion : QuestionBase
+    {
+        [Required]
+        public ICollection<SingleMultipleAnswerQuestionOption> SingleMutipleAnswerQuestionOption { get; set; }
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/SingleMultipleAnswerQuestion.cs]: why is the virtual keyword missing?

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/SingleMultipleAnswerQuestionOption.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+namespace Promact.Trappist.DomainModel.Models.Question
+{
+    public class SingleMultipleAnswerQuestionOption:BaseModel
+    {
+        [Required]
+        public string Option { get; set; }
+        public bool IsAnswer { get; set; }
+        [ForeignKey("Question_Id")]
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/SingleMultipleAnswerQuestionOption.cs]: Where is Question_Id field?

File: Trappist/src/Promact.Trappist.Web/Migrations/20170309072938_model change.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Promact.Trappist.Web.Migrations
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/Migrations/20170309072938_model change.cs]: why this migration file is empty?

Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/Migrations/20170309072938_model change.cs]: why is this migration file empty?

File: Trappist/src/Promact.Trappist.Web/Migrations/20170309074509_model name change.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Promact.Trappist.Web.Migrations
+{
+    public partial class modelnamechange : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/Migrations/20170309074509_model name change.cs]: why there is so many migration files exist?

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionBase.cs (hunk 3 above)
Comment (viharsanghavi) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionBase.cs]: It refers user id of test conductor

Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionBase.cs]: But why it is int? It suppose to be string.

Comment (viharsanghavi) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionBase.cs]: Ok I got it your point. I will change it to string.

File: Trappist/global.json (hunk 1 above)
Comment (viharsanghavi) [inline Trappist/global.json]: In my system older version is there so i change it to older version

Comment (Krunal-Promact) [inline Trappist/global.json]: If there is an older version in your system then you suppose to update in your system, not in the code

Comment (viharsanghavi) [inline Trappist/global.json]: ok
